### PR TITLE
Fix image provider lifecycle

### LINKED ISSUE #341: Exception occurs.
**Describe the bug**
I'm passing the url from one screen to another and trying to show the image with Photo_View, but keep getting the exception. 
```
Exception has occurred.
_AssertionError ('package:flutter/src/painting/image_stream.dart': Failed assertion: line 245 pos 12: '_completer == null': is not true.)
```

**To Reproduce**
Pass a url String in arguments of Navigator.pushNamed from one screen to another. 

Class that shows the image:

```
class _ImageZoomState extends State<ImageZoom> {
  String url;

  @override
  Widget build(BuildContext context) {
    Map data = ModalRoute.of(context).settings.arguments;
    url = data['url'];
    return Scaffold(
      appBar: AppBar(
        title: Text('Image'),
      ),
      body: Container(
          child: Center(child: PhotoView(imageProvider: NetworkImage(url)))),
    );
  }
}

```

**What is the current behavior?**
Exception occurs. 
**Expected behavior**
Should show a network image. 
**Which versions of Flutter/Photo View, and which browser / OS are affected by this issue? Did this work in previous versions of Photo View?**
Flutter doctor:
[✓] Flutter (Channel master, 1.22.0-10.0.pre.200, on Mac OS X 10.15.4 19E287, locale en-KG)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
[✓] Android Studio (version 4.0)
[✓] Connected device (2 available)

• No issues found!

### PR DESCRIPTION
**Pull request by renancaraujo**
This may fix #341

Added recycling mechanics for the internal image stream.

### CODE REVIEW

File: lib/src/photo_view_wrappers.dart [hunk 1]
```diff
@@ -113,19 +116,39 @@ class _ImageWrapperState extends State<ImageWrapper> {
...
       onError: handleError,
     );
-    _stream.addListener(_imageStreamListener);
+
+    return _imageStreamListener;
+  }
+
+  void _updateSourceStream(ImageStream newStream) {
+    if (_imageStream != null) {
+      if (_imageStream?.key == newStream.key) {
```
Comment (erickzanardo) [inline lib/src/photo_view_wrappers.dart:125]: You already now that this isn't null

**Review by erickzanardo — APPROVED** [review]
One bit, other than that LGTM

File: lib/src/photo_view_wrappers.dart (hunk 1 above)
Comment (renancaraujo) [reply lib/src/photo_view_wrappers.dart:125]: `didUpdateWidget` is called after `initState`, where the stream can be null.

Comment (erickzanardo) [reply lib/src/photo_view_wrappers.dart:125]: But this statement is inside an if that check if the variable isn't null, no need for the Elvis operator

Comment (luanpotter) [reply lib/src/photo_view_wrappers.dart:125]: better yet -- if you know `newStream.key`, the condition `_imageStream?.key == newStream.key` is always false for null _imageStream.

Comment (renancaraujo) [reply lib/src/photo_view_wrappers.dart:125]: heck, forgot that.

Comment (renancaraujo) [reply lib/src/photo_view_wrappers.dart:125]: To be fixed on #371

**Pull request merged**